apply/golden_file_test.go

```diff
@@ -62,20 +63,37 @@ func TestIntegration(t *testing.T) {
 				e = apply.Apply(testdataFs, conf, templates.Templates, true)
 				a.NoError(e)
 			} else {
-
+				isYaml := true
 				fs, _, e := util.TestFs()
 				a.NoError(e)
 
 				// copy fogg.json into the tmp test dir (so that it doesn't show up as a diff)
-				configContents, e := afero.ReadFile(testdataFs, "fogg.json")
-				a.NoError(e)
-				configMode, e := testdataFs.Stat("fogg.json")
+				configContents, e := afero.ReadFile(testdataFs, "fogg.yml")
+				if os.IsNotExist(e) {
+					configContents, e = afero.ReadFile(testdataFs, "fogg.json")
+					isYaml = false
+				}
 				a.NoError(e)
-				a.NoError(afero.WriteFile(fs, "fogg.json", configContents, configMode.Mode()))
 
-				conf, e := config.FindAndReadConfig(fs, "fogg.json")
-				a.NoError(e)
-				fmt.Printf("conf %#v\n", conf)
+				var configMode os.FileInfo
+				var conf *v2.Config
+				if isYaml {
+					configMode, e = testdataFs.Stat("fogg.yml")
```

Comment:
the only thing that changes between this conditional and the other is `fogg.{json|yml}`, maybe you can do something like:
```
filename := "fogg.yml"
if !isYaml {
filename = "fogg.json"
}

// So all of this code is not duplicated anymore
configMode, e = testdataFs.Stat(filename)
....

```